.github/workflows/build.yml

```
@@ -80,6 +101,18 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v4.2.1
+      - name: Configure secrets.yaml file
+        # if: inputs.secrets-yaml-file != ''
+        # if: secrets.ESPHOME_SECRETS_YAML != ''
+        uses: akiojin/decode-base64-github-action@v1.0.2
+        id: decode-base64
+        with:
+          base64: ${{ secrets.ESPHOME_SECRETS_YAML }}
+          output-path: ${{ inputs.secrets-yaml-file-path }}
```

Probably don't need to import a whole action for this
```suggestion
        id: decode-base64
        run: echo "${{ secrets.ESPHOME_SECRETS_YAML }}" | base64 -d > ${{ inputs.secrets-yaml-file-path }}
```

---

👍